code/modules/mob/living/carbon/human/say.dm

```diff
@@ -16,6 +16,10 @@
 				return real_name
 		else
 			return real_name
+	if(istype(wear_mask, /obj/item/clothing/mask/infiltrator))
+		var/obj/item/clothing/mask/infiltrator/V = wear_mask
+		if(V.vunknown)
+			return ("Unknown")

```

Comment:
This is a property pretty similar to the chameleon's `vchange` and could be generalized as a mask property.
As in moving both variables to the mask level (maybe renaming it to something more clear like `voice_change` and `voice_unknown`) and performing a single `istype` check here. More generic behavior.